src/duckling/game/drawable/tile-block-drawable-drawer.ts

```diff
@@ -0,0 +1,130 @@
+import {RenderTexture, Texture, Sprite, Graphics, DisplayObject, Container, BaseTexture} from 'pixi.js';
+
+import {Vector} from '../../math/vector';
+import {AssetService} from '../../project/asset.service';
+
+import {TileBlockDrawable, getTileWidth} from './tile-block-drawable';
+
+export function drawTileBlockDrawable(tileBlockDrawable : TileBlockDrawable, assetService : AssetService) : DisplayObject {
+    if (!tileBlockDrawable.textureKey) {
+        return null;
+    }
+
+    let baseTexture = assetService.get({key: tileBlockDrawable.textureKey, type: "TexturePNG"});
+    if (!baseTexture) {
+        return null;
+    }
+    
+    let container = new Container();
+    container.addChild(_constructTileBlockSprite(tileBlockDrawable, baseTexture, assetService));
+    return container;
+}
+
+function _constructTileBlockSprite(tileBlockDrawable : TileBlockDrawable, baseTexture : BaseTexture, assetService : AssetService) : DisplayObject {
+    let tileSize = getTileWidth(tileBlockDrawable, assetService);
+    let tiles = { x: tileBlockDrawable.size.x / tileSize, y: tileBlockDrawable.size.y / tileSize};
+    
+    let positions = _getPositions(tiles);
+    let calculateOffset: {x: boolean, y: boolean} = _getOffsets(tiles);
+    
+    return _constructTileBlockContainer(positions, calculateOffset, baseTexture, tileSize, tiles);
+}
+
+function _getPositions(tiles: Vector): Vector[] {
+    let positions: Vector[] = [];
+    
+    let xPositions: number[] = _getAxisPositions(tiles.x);
+    let yPositions: number[] = _getAxisPositions(tiles.y);
+    
+    let position: Vector;
+    for (let xPosition of xPositions) {
+        for (let yPosition of yPositions) {
+            position = { x: xPosition, y: yPosition };
+            positions.push(position);
+        }
+    }
+
+    return positions;
+}
+
+function _getAxisPositions(numTilesOnAxis: number): number[] {
+    if (numTilesOnAxis > 1) {
+        return [0,1,2];
+    } else {
+        return [3];
+    }
+}
+
+function _getOffsets(tiles: Vector): { x : boolean, y: boolean } {
+    let calculateOffset: { x: boolean, y: boolean } = { x: true, y: true };
+    if (tiles.x === 1) {
+        calculateOffset.x = false;
+    } 
+    if (tiles.y === 1) {
+        calculateOffset.y = false;
+    } 
+    return calculateOffset;
+}
+
+
+function _constructTileBlockContainer(positions: Vector[], calculateOffset: { x: boolean, y: boolean}, baseTexture: BaseTexture, tileSize: number, tiles: Vector): Container {
+    let texture: Texture;
```

Comment:
Since the texture and sprite variables are only used in the loop you should move the declarations into the loop.